### ISSUE TITLE
Combobox: how can the user clear/reset a fabric ui

### ISSUE THREAD

**Issue by petercutting** [OfficeDev/office-ui-fabric-react#5917]
A user can select an option in a combobox, but there seems to be no obvious way to clear/reset a previous selection

Here are some possibilities for how the component could support this requirement:
* keyboard, eg DEL and/or BackSpace
* reset option at top of dropdown list
* small docked button.

Why is there no support in the component? Why is there nothing in the documentation to atleast mention that there is no support and maybe what options the progammer has to workaround 👎 

workarounds seem to be
* handling keyboard events - advanced
* manually add reset option to the list and handle it 
* allowFreeform configuration - suboptimal if you dont want user typing in field

What am I missing?

Peter

**Comment by KevinTCoughlin** [comment]
@petercutting would you be able to extend the component in a way similar to [this Codepen](https://codepen.io/kevintcoughlin/pen/ajMrmm) to unblock yourself since there isn't currently a first class `reset` method on `ComboBox`?

**Comment by cliffkoh** [comment]
@JasonGore @KevinTCoughlin @micahgodbolt is this still correctly tagged as a question? Or is it related to #5929 as a verified ComboBox bug?

**Comment by micahgodbolt** [comment]
Looks like the request is to be able to remove freeform additions

![image](https://user-images.githubusercontent.com/1434956/46830560-cf61af80-cd55-11e8-9fe7-439927ee5851.png)

Would be nice to have an 'x' next to that, at least on focus/hover, to be able to remove it

**Comment by micahgodbolt** [comment]
@betrue-final-final have you seen any examples of dropdown lists where you can remove items from them? In this case we'd want to be able to remove options that we added.

**Comment by betrue-final-final** [comment]
I'm sure I have somewhere, but would need to look to find them. I would suspect that yes an "x" on the right would appear on focus or hover. Of course this will complicate the keyboarding slightly, but a right arrow could expose the delete, and using the delete key should also work.

**Comment by micahgodbolt** [comment]
having the X would help indicate it could be deleted as well

**Comment by betrue-final-final** [comment]
I would only expose it on hover or focus.

**Comment by betrue-final-final** [comment]
Wait, this isn't about removing an option, but just deselection. Here's the original text: "A user can select an option in a combobox, but there seems to be no obvious way to clear/reset a previous selection."

In most cases, they are selecting another option. I think in this case, he wants to zero out the dropdown back to no selection at all. One way that VSO does that is to have a clear option at bottom. That seems much for a default though. I'm not sure if this is a problem that needs solving though.

**Comment by betrue-final-final** [comment]
We were just talking about this. Yeah, I wouldn't expose this normally as it's not something dropdowns have ever really done and it's kind of a niche. THe other way is to have an additional menu item to "Clear selection". VSTS does this in their filters. That way you won't think it's search.

**Comment by micahgodbolt** [comment]
oh that's a decent idea. should that be baked in, or just mention it as a good UI practice if necessary.

**Comment by jspurlin** [comment]
Yeah, I agree either add an option to clear the value or someone could create a button next to the combobox (really anywhere they want) which clears the value (via sending in new props to clear it). This is something that I feel should be handled on a case by case basis and not be baked in to the component

**Comment by micahgodbolt** [comment]
Okay, so the resolution is: Best practice is to simply have an option in the dropdown, or button next to the combobox with an onClick that sets the combobox 'text' to an empty string.

**Comment by jspurlin** [comment]
Yep, sounds good to me. If you want the "clear" functionality, that should be the best practice to achieve it

**Comment by petercutting** [comment]
Keyboard access hasnt been mentioned. So I just suggest that resetting the selection should be possible using just the keyboard (ie not mouse)

**Comment by micahgodbolt** [comment]
a 'clear' button next to the combobox is a tab away, just like a button placed inside of the combobox. onClick could even return focus to the comobobox if you wanted to.
